views/components/App.jsx

```diff
@@ -7,12 +7,17 @@ export default class App extends React.Component {
     super(props);
     // Binding makes the correct 'this' reference in the callback's context
     this.selectOutput = this.selectOutput.bind(this);
+    this.selectInput = this.selectInput.bind(this);
```

Comment:
so this is an annoying pattern (needing to bind event listener methods); I prefer letting babel help.

When defining `selectInput`, don't do this:
```
selectInput () {
  /* do stuff */
}
```
instead do this
```
selectInput = () => {
  /* do stuff */
}
```
This is equivalent to binding `this` in the constructor.